tests/test_client.py

```diff
@@ -225,15 +226,56 @@ def test_connect_close(self):
         self.sg.close()
         self.assertEqual(None, self.sg._connection)
 
-
     def test_network_retry(self):
-        """Network failure is retried"""
+        """Network failure is retried, with a sleep call between retries."""
         self.sg._http_request.side_effect = httplib2.HttpLib2Error
 
-        self.assertRaises(httplib2.HttpLib2Error, self.sg.info)
-        self.assertTrue(
-            self.sg.config.max_rpc_attempts ==self.sg._http_request.call_count,
-            "Call is repeated")
+        with mock.patch("time.sleep") as mock_sleep:
+            self.assertRaises(httplib2.HttpLib2Error, self.sg.info)
+            self.assertTrue(
+                self.sg.config.max_rpc_attempts == self.sg._http_request.call_count,
+                "Call is repeated")
+            # Ensure that sleep was called with the retry interval between each attempt
+            attempt_interval = self.sg.config.rpc_attempt_interval / 1000.0
+            calls = [mock.callargs(((attempt_interval,), {}))]
+            calls *= (self.sg.config.max_rpc_attempts - 1)
+            self.assertTrue(
+                mock_sleep.call_args_list == calls,
+                "Call is repeated at correct interval."
+            )
+
+    def test_set_retry_interval(self):
+        """Setting the retry interval through parameter and environment variable works."""
+        original_env_val = os.environ.pop("SHOTGUN_API_RETRY_INTERVAL", None)
```

Comment:
If your test fails then the environment variable state will leak into other tests.

Maybe you could pass the value the environment variable should be set to inside `run_internal_test` and inside `run_interval_test` have something like
```
old_value =  os.environ.pop("SHOTGUN_API_RETRY_INTERVAL")
os.environ["SHOTGUN_API_RETRY_INTERVAL"] = new_retry_interval
try:
   # your test code goes here.
finally:
   if old_value is not None:
      os.environ["SHOTGUN_API_RETRY_INTERVAL"] = old_value
```

Reply:
✅wrapped in try-finally, and also ensure we won't try to pop from os.environ if the key doesn't exist.